### DISCUSSION TITLE
Training Onnx Models

### DISCUSSION THREAD

**Discussion by IzanCatalan** [microsoft/onnxruntime#15868] [Other Q&A]
Hi everyone!

I would like to know if it is possible to train Neural Network models using Onnx Runtime and export them in ONNX format.

Moreover, I wonder if it is also possible to re-train some pre-trained, state-of-the-art ONNX format models (https://github.com/onnx/models) in order to do Transfer Learning. Any help would be appreciated. Thanks!

Izan.

**Comment by AdamLouly** [comment]
Hey IzanCatalan,
If you want to train models using ONNXRuntime, you can refer to [ort_module](https://github.com/microsoft/onnxruntime-training-examples)
https://github.com/pytorch/ort
If you wish to directly use ONNXRuntime apis (C, C++ etc), training APIs provided by ONNX Runtime are primarily intended for on-device training scenarios or specific constraints.
you can check more examples here: https://github.com/microsoft/onnxruntime-training-examples/tree/master/on_device_training
